Mathlib/Data/Multiset/Fintype.lean

```diff
@@ -236,55 +232,56 @@ theorem Multiset.image_toEnumFinset_fst (m : Multiset α) :
 #align multiset.image_to_enum_finset_fst Multiset.image_toEnumFinset_fst
 
 @[simp]
-theorem Multiset.map_univ_coe (m : Multiset α) : (Finset.univ : Finset m).val.map coe = m := by
-  have := m.map_to_enum_finset_fst
-  rw [← m.map_univ_coe_embedding] at this
+theorem Multiset.map_univ_coe (m : Multiset α) :
+    (Finset.univ : Finset m).val.map (fun x : m ↦ (x : α)) = m := by
+  have := m.map_toEnumFinset_fst
+  rw [← m.map_univ_coeEmbedding] at this
   simpa only [Finset.map_val, Multiset.coeEmbedding_apply, Multiset.map_map,
     Function.comp_apply] using this
 #align multiset.map_univ_coe Multiset.map_univ_coe
 
 @[simp]
 theorem Multiset.map_univ {β : Type _} (m : Multiset α) (f : α → β) :
-    ((Finset.univ : Finset m).val.map fun x => f x) = m.map f := by
-  rw [← Multiset.map_map, Multiset.map_univ_coe]
+    ((Finset.univ : Finset m).val.map fun (x : m) ↦ f (x : α)) = m.map f := by
+  erw [← Multiset.map_map]
+  rw [Multiset.map_univ_coe]
 #align multiset.map_univ Multiset.map_univ
 
 @[simp]
-theorem Multiset.card_toEnumFinset (m : Multiset α) : m.toEnumFinset.card = m.card := by
-  change Multiset.card _ = _
-  convert_to (m.to_enum_finset.val.map Prod.fst).card = _
-  · rw [Multiset.card_map]
-  · rw [m.map_to_enum_finset_fst]
+theorem Multiset.card_toEnumFinset (m : Multiset α) : m.toEnumFinset.card = Multiset.card m := by
+  rw [Finset.card, ←Multiset.card_map Prod.fst m.toEnumFinset.val]
+  congr
+  exact m.map_toEnumFinset_fst
 #align multiset.card_to_enum_finset Multiset.card_toEnumFinset
 
 @[simp]
-theorem Multiset.card_coe (m : Multiset α) : Fintype.card m = m.card := by
-  rw [Fintype.card_congr m.coe_equiv]
-  simp
+theorem Multiset.card_coe (m : Multiset α) : Fintype.card m = Multiset.card m := by
+  rw [Fintype.card_congr m.coeEquiv]
+  simp only [Fintype.card_coe, card_toEnumFinset]
 #align multiset.card_coe Multiset.card_coe
 
 @[to_additive]
-theorem Multiset.prod_eq_prod_coe [CommMonoid α] (m : Multiset α) : m.Prod = ∏ x : m, x := by
+theorem Multiset.prod_eq_prod_coe [CommMonoid α] (m : Multiset α) : m.prod = ∏ x : m, (x : α) := by
   congr
-  simp
+  -- Porting note: `simp` fails with "maximum recursion depth has been reached"
+  erw [map_univ_coe]
 #align multiset.prod_eq_prod_coe Multiset.prod_eq_prod_coe
-#align multiset.sum_eq_sum_coe Multiset.sum_eq_sum_coe
+#noalign multiset.sum_eq_sum_coe
```

Comment:
Why are you `noalign`ing these? they should be aligned to the additive version, as mathport suggests.